### ISSUE TITLE
`--global`, `--local` are deprecated

### ISSUE THREAD

**Issue by raineorshine** [actions/setup-node#515]
**Description:**
On Windows Node v18.3.0, every time `npm` is spawned, I get the following warning printed to `stderr`:

```
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
```

This occurs not just when installing dependencies and running tests within a job, but also whenever `npm` is spawned within any of those tests. This is causing a failure in `npm-check-updates` tests that rely on `stderr` to be completely empty: https://github.com/raineorshine/npm-check-updates/runs/6850288596?check_suite_focus=true#step:6:471.

**Action version:**
Specify the action version: `v3.1.0`

**Platform:**
- [ ] Ubuntu
- [ ] macOS
- [x] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**

node v18.3.0

**Repro steps:**  
A description with steps to reproduce the issue. If you have a public example or repo to share, please provide the link.

https://github.com/raineorshine/npm-check-updates/runs/6850288596?check_suite_focus=true#step:6:471

**Expected behavior:**
On Windows 18, `npm` should be configured to not use the deprecated `--global` and `--local` options, and no warning should be shown.

**Actual behavior:**
The warning is printed to `stderr` any time `npm` is run on the host machine:

```
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
```

In the [failed Github action](https://github.com/raineorshine/npm-check-updates/runs/6850288596?check_suite_focus=true#step:6:471), you can see the warnings printed in both setup and npm install, before any user tests are run:

```sh
Run actions/setup-node@v3.1.0
Attempting to download 18...
Acquiring 18.3.0 - x64 from https://github.com/actions/node-versions/releases/download/18.3.0-2426429734/node-18.3.0-win32-x64.7z
Extracting ...
Adding to the cache ...
Done
C:\Windows\system32\cmd.exe /D /S /C "C:\hostedtoolcache\windows\node\18.3.0\x64\npm.cmd config get cache"
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
C:\npm\cache
npm cache is not found
```

During `npm install`:

```sh
Run npm ci
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
```

**Notes:**

- I have searched existing issues
- I have confirmed that all instances of `--global` and `--local` have been removed from the `npm-check-updates` codebase. It seems to be coming from the npm config installed by `setup-node`.
- The issue does not occur on Ubuntu.

**Comment by IvanZosimov** [comment]
Hi, @raineorshine 👋 ! Thank you for the issue. Here you encountered the [well known bug](https://github.com/npm/cli/issues/4980) of npm. At that moment this bug is fixed in npm 8.12.1, but this version still isn't bundled with the LTS and Current (16.x and 18.x) versions of the node.js. As soon as the mentioned versions of node.js get updated, your workflow will start to perform normally. By now, I suggest you to use this workaround which can be inserted into your workflow right after the setup-node step.
```- name: Update bundled npm version
        if: ${{ (matrix.node == '16' || matrix.node == '18') && matrix.os == 'windows-latest' }}
        run: |
           $WhereNode = Get-Command node | Select-Object -ExpandProperty Definition
           $NodeDirPath = Split-Path $WhereNode -Parent
           cd $NodeDirPath
           npm install npm@8.12.1
```
Don't forget to delete it after the new version of node.js is revealed.

By now, I'm closing this issue, If you have any questions or suggestions feel free to ping us.

**Comment by raineorshine** [comment]
Great, thank you for the explanation and workaround!
